### ISSUE TITLE
Default limit value is not applied on searches

### ISSUE THREAD

**Issue by edmundoa** [Graylog2/graylog2-server#1775]
Using the current master, the default limit parameter for searches is not working. Here's the result of the query `http://hostname:12900/search/universal/relative?query=*&range=3600&pretty=true`:

```
{
  "query" : "*",
  "built_query" : "{\"from\":0,\"query\":{\"bool\":{\"must\":{\"match_all\":{}},\"filter\":{\"bool\":{\"must\":{\"range\":{\"timestamp\":{\"from\":\"2016-02-05 17:25:44.854\",\"to\":\"2016-02-05 18:25:44.854\",\"include_lower\":true,\"include_upper\":true}}}}}}},\"sort\":[{\"timestamp\":{\"order\":\"desc\"}}],\"highlight\":{\"require_field_match\":false,\"fields\":{\"*\":{\"fragment_size\":0,\"number_of_fragments\":0}}}}",
  "used_indices" : [ {
    "index_name" : "graylog2_19",
    "begin" : "1970-01-01T00:00:00.000Z",
    "end" : "1970-01-01T00:00:00.000Z",
    "calculated_at" : "2016-02-05T13:28:28.379Z",
    "took_ms" : 9
  }, {
    "index_name" : "graylog2_5",
    "begin" : "2015-12-17T10:34:30.794Z",
    "end" : "2016-12-17T17:26:41.000Z",
    "calculated_at" : "2016-01-26T14:11:56.019Z",
    "took_ms" : 13
  } ],
  "messages" : [ {
    "message" : {},
    "index" : "graylog2_19",
    "highlight_ranges" : null
  }, {
    "message" : {},
    "index" : "graylog2_19",
    "highlight_ranges" : null
  }, {
    "message" : {},
    "index" : "graylog2_19",
    "highlight_ranges" : null
  }, {
    "message" : {},
    "index" : "graylog2_19",
    "highlight_ranges" : null
  }, {
    "message" : {},
    "index" : "graylog2_19",
    "highlight_ranges" : null
  }, {
    "message" : {},
    "index" : "graylog2_19",
    "highlight_ranges" : null
  }, {
    "message" : {},
    "index" : "graylog2_19",
    "highlight_ranges" : null
  }, {
    "message" : {},
    "index" : "graylog2_19",
    "highlight_ranges" : null
  }, {
    "message" : {},
    "index" : "graylog2_19",
    "highlight_ranges" : null
  }, {
    "message" : {},
    "index" : "graylog2_19",
    "highlight_ranges" : null
  } ],
  "fields" : [],
  "time" : 6,
  "total_results" : 23914,
  "from" : "2016-02-05T17:25:44.861Z",
  "to" : "2016-02-05T18:25:44.861Z"
}
```

As you can see, the search is returning 10 results instead of 150. Looking a bit through the code, I saw that the limit applied in `SearchesConfig.Builder` is overrode by setting a 0 `limit` afterwards, and that is happening in some places where we rely on the default limit.

**Comment by kroepke** [comment]
hm, why isn't this issue fixed? the PR seems to be merged? @edmundoa @joschi ?

**Comment by joschi** [comment]
@kroepke The "magic keywords" in a2e03da96ffd6ae4e2b98d275960989ba072060a were wrong ("Fixes issue #1775" instead of "Fixes #1775").

**Comment by kroepke** [comment]
ah duh :)
